leptos_macro/src/lib.rs

```diff
@@ -644,6 +645,128 @@ pub fn component(args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
         .into()
 }
 
+/// Annotates a struct so that it can be used with your Component as a `slot`.
+///
+/// The `#[slot]` macro allows you to annotate plain Rust struct as component slots and use them
+/// within your Leptos [component](crate::component!) properties. The struct can contain any number
+/// of fields. When you use the component somewhere else, the names of the slot fields are the
+/// names of the properties you use in the [view](crate::view!) macro.
+///
+/// Here’s how you would define and use a simple Leptos component which can accept a custom slot:
+/// ```rust
+/// # use leptos::*;
+/// use std::time::Duration;
+///
+/// #[slot]
+/// struct HelloSlot {
+///     // Same prop syntax as components.
+///     #[prop(optional)]
+///     children: Option<Children>,
+/// }
+///
+/// #[component]
+/// fn HelloComponent(
+///     cx: Scope,
+///     /// Component slot, should be passed through the <HelloSlot slot> syntax.
+///     hello_slot: HelloSlot,
+/// ) -> impl IntoView {
+///     // mirror the children from the slot, if any were passed
+///     if let Some(children) = hello_slot.children {
+///         (children)(cx).into_view(cx)
+///     } else {
+///         ().into_view(cx)
+///     }
+/// }
+///
+/// #[component]
+/// fn App(cx: Scope) -> impl IntoView {
+///     view! { cx,
+///         <HelloComponent>
+///             <HelloSlot slot>
+///                 "Hello, World!"
+///             </HelloSlot>
+///         </HelloComponent>
+///     }
+/// }
+/// ```
+///
+/// /// Here are some important details about how slots work within the framework:
+/// 1. Most of the same rules from [component](crate::component!) macro should also be followed on slots.
+///
+/// 2. Specifying only `slot` without a name (such as in `<HelloSlot slot>`) will default the chosen slot to
+/// the a snake case version of the slot struct name (`hello_slot` for `<HelloSlot>`).
+///
+/// 3. Currently, event handlers cannot be specified directly on the slot.
+///
+/// ```compile_error
+/// // ❌ This won't work
+/// # use leptos::*;
+///
+/// #[slot]
+/// struct SlotWithChildren {
+///     children: Children,
+/// }
+///
+/// #[component]
+/// fn ComponentWithSlot(cx: Scope, slot: SlotWithChildren) -> impl IntoView {
+///     (slot.children)(cx)
+/// }
+///
+/// #[component]
+/// fn App(cx: Scope) -> impl IntoView {
+///     view! { cx,
+///         <ComponentWithSlot>
+///           <SlotWithChildren slot:slot on:click=move |_| {}>
+///             <h1>"Hello, World!"</h1>
+///           </SlotWithChildren>
+///         </ComponentWithSlot>
+///     }
+/// }
+/// ```
+///
+/// ```
+/// // ✅ Until fixed, do this instead
```

Comment:
FWIW I think this is probably the correct behavior and not something that needs to be fixed: If a "slot" is defined as a way of building a struct and passing it as one of the props to another component, I can't imagine what it would mean to add an event listener to it, except in the specific case of using a slot to pass through `children`, in which case — because `children` can return, for example, a fragment or nothing — the event listener either needs to be added on an element inside the slot, or needs to be added by the parent component that takes the slot.

Reply:
I believe this is 100% correct. The parent component should be responsible for adding events to components.